src/panels/lovelace/cards/hui-clock-card.ts

```diff
@@ -0,0 +1,244 @@
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import "../../../components/ha-alert";
+import "../../../components/ha-card";
+import type { HomeAssistant } from "../../../types";
+import type {
+  LovelaceCard,
+  LovelaceCardEditor,
+  LovelaceGridOptions,
+} from "../types";
+import type { ClockCardConfig } from "./types";
+import { TimeFormat } from "../../../data/translation";
+import { useAmPm } from "../../../common/datetime/use_am_pm";
+import { resolveTimeZone } from "../../../common/datetime/resolve-time-zone";
+
+const INTERVAL = 1000;
+
+@customElement("hui-clock-card")
+export class HuiClockCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import("../editor/config-elements/hui-clock-card-editor");
+    return document.createElement("hui-clock-card-editor");
+  }
+
+  public static getStubConfig(): ClockCardConfig {
+    return {
+      type: "clock",
+      clock_size: "medium",
+      show_seconds: true,
+      time_format: TimeFormat.am_pm,
+    };
+  }
+
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @state() private _config?: ClockCardConfig;
+
+  @state() private _date?: Intl.DateTimeFormat;
+
+  @state() private _timeHour?: string = "00";
+
+  @state() private _timeMinute?: string = "00";
+
+  @state() private _timeSecond?: string = "00";
+
+  @state() private _timeAmPm?: string = "--";
+
+  private _tickInterval?: undefined | number;
+
+  public setConfig(config: ClockCardConfig): void {
+    this._config = config;
+    this._initDate();
+  }
+
+  private _initDate() {
+    let locale = this.hass?.locale;
+
+    if (!locale || !this.hass) return;
+
+    if (this._config?.time_format) {
+      locale = { ...locale, time_format: this._config.time_format };
+    }
+
+    this._date = new Intl.DateTimeFormat(this.hass?.locale?.language, {
```

Comment:
You already checked that hass and locale are not undefined
```suggestion
    this._date = new Intl.DateTimeFormat(this.hass.locale.language, {
```